wallet-new/src/Cardano/Wallet/WalletLayer/Types.hs

```diff
@@ -14,39 +18,54 @@ module Cardano.Wallet.WalletLayer.Types
     , updateAccount
     , deleteAccount
 
+    , createAddress
     , getAddresses
+    , isAddressValid
     ) where
 
+
 import           Universum
 
 import           Control.Lens (makeLenses)
+import           Control.Exception.Safe (try)
 
-import           Cardano.Wallet.API.V1.Types (Account, AccountIndex, AccountUpdate, Address,
-                                              NewAccount, NewWallet, Wallet, WalletId, WalletUpdate)
+import           Cardano.Wallet.API.V1.Types (Account, AccountIndex, AccountUpdate, AddressValidity,
+                                              NewAccount, NewAddress, NewWallet, Wallet,
+                                              WalletAddress, WalletId, WalletUpdate)
 
 import           Cardano.Wallet.Kernel.Diffusion (WalletDiffusion (..))
+import           Cardano.Wallet.WalletLayer.Error (WalletLayerError (..))
 
 ------------------------------------------------------------
 -- Passive wallet layer
 ------------------------------------------------------------
 
+-- The response for the @WalletPassiveLayer@.
+type WalletLayerResponse m a = m (Either WalletLayerError a)
+
+-- A utility function for transforming between representations.
+monadThrowToEither :: forall m a. (MonadCatch m) => m a -> WalletLayerResponse m a
+monadThrowToEither = try
```

Comment:
When we're working with the `PassiveWalletLayer`, do we expect every single function to potentially fail?

When I was implementing the HTTP client, every request is going over the network, which does have an inherent failure possibility. I may be mistaken, but these functions are local, and that suggests to me that they should mostly succeed, except with specific error modes as reflected in the type.

Reply:
It's reading from a file, so I think we can say it's not really stable and none of them are safe.

Reply:
Sounds good :)